### ISSUE TITLE
Data loss after restarting chunkserver: chunks files lost (removed) due to change in "number of total blocks"

### ISSUE THREAD

**Issue by onlyjob** [moosefs/moosefs#207]
I've just got a first incident of data loss on MooseFS. As an experiment, I've configured chunkserver to keep chunks to (healthy) FUSE file system. Shortly I've found chunkserver storage marked as "damaged":

~~~~
disk: /mnt/r/.lfs/ ; number of total blocks has been changed significantly (1728694097 -> 1200168102) - mark it as damaged
~~~~

Fair enough: I've marked path in `mfshdd.cfg` as follows: `~/mnt/r/.lfs` to "ignore significant change of hard drive total size".

After restarting, chunkserver failed immediately, marking storage as "damaged":

~~~~
scanning folder /mnt/r/.lfs/: complete (3s)
hdd_io_begin: file:/mnt/r/.lfs//00/chunk_00000000023DA821_00000002.mfs - open error: ENOENT (No such file or directory)
: hdd_io_begin: file:/mnt/r/.lfs//00/chunk_00000000023DA821_00000002.mfs - open error: ENOENT (No such file or directory)
~~~~

I've restarted chunkserver once again:

~~~~
scanning folder /mnt/r/.lfs/: .chunkdb used - full scan don't needed
scanning folder /mnt/r/.lfs/: complete (1s)
hdd_io_begin: file:/mnt/r/.lfs//00/chunk_00000000023DD7AC_00000001.mfs - open error: ENOENT (No such file or directory)
hdd_io_begin: file:/mnt/r/.lfs//00/chunk_000000000245A9DE_00000001.mfs - open error: ENOENT (No such file or directory)
hdd_io_begin: file:/mnt/r/.lfs//00/chunk_00000000023DE077_00000001.mfs - open error: ENOENT (No such file or directory)
hdd_io_begin: file:/mnt/r/.lfs//00/chunk_00000000023E4D57_00000001.mfs - open error: ENOENT (No such file or directory)
4 errors occurred in 600 seconds on folder: /mnt/r/.lfs/
~~~~

Between restarts I've noticed that chunks files in `/mnt/r/.lfs/00` disappeared.

Result: 1151 missing chunks, just by restarting chunkserver few times.
Chunks that were successfully saved were removed by chunkserver itself. 

Apparently, something went badly wrong with saving list of chunks into `.chunkdb` after when chunkserver marked file system as "damaged". I think `.chunkdb` was empty when chunkserver has been restarted so chunkserver removed all the valid chunk files on start.

This is a serious bug that can easily cause severe data loss (as it did).
Chunkserver should not rely on `.chunkdb` file from "damaged" file system. 

Removal of chunks files were not even logged...

Also it might be useful to log total number of known chunks after when chunkserver finished scanning its file systems or finished loading list of chunks from `.chunkdb`.

**Comment by onlyjob** [comment]
Basically I've lost all chunk/files created prior to adding `~` in front of storage path in `mfshdd.cfg`...

**Comment by borkd** [comment]
Out of curiosity - what replication level did you set for any lost files/chunks? One?

**Comment by onlyjob** [comment]
@borkd, this issue is about loss of chunk files. Replication level is not relevant as software bug that causes loss of all chunksever's chunk files can easily destroy more than one replica when few affected chunkservers are restarted.

**Comment by onlyjob** [comment]
Mich better now -- thank you. Perhaps this issue is fixed.

IMHO this fix alone justifies releasing v3.0.104.

**Comment by oxide94** [comment]
@onlyjob we are planning to do it very soon, hopefully this week.
Piotr
